File: src/main/java/org/glavo/viewer/gui/ViewerPane.java
package org.glavo.viewer.gui;

import javafx.scene.layout.BorderPane;
import org.glavo.viewer.util.CssUtils;
import org.jetbrains.annotations.NotNull;

public final class ViewerPane extends BorderPane {
    @NotNull
    public final Viewer viewer;

    public ViewerPane(@NotNull Viewer viewer) {
        this.viewer = viewer;
        this.getStyleClass().add(CssUtils.UI_CLASS);
        this.setTop(new ViewerTopBar(viewer));
    }


    @NotNull
    public ViewerTopBar getTopBar() {
        return (ViewerTopBar) getTop();
    }
}

File: src/main/kotlin/org/glavo/viewer/gui/ViewerPane.kt
package org.glavo.viewer.gui

import javafx.scene.layout.BorderPane

class ViewerPane(val viewer: Viewer) : BorderPane() {
    init {
        top = ViewerTopBar(viewer)
    }

    val topBar: ViewerTopBar
        inline get() = this.top as ViewerTopBar
}